snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/SnapshotSaver.kt

```diff
@@ -105,17 +118,30 @@ internal object SnapshotSaver {
    * We intentionally don't account for FQN here as we still want to diff an image if
    * the test might move packages, which user-defined name ensures.
    */
-  private fun keyName(displayName: String): String {
-    // Replace spaces & periods with underscores and lowercase the string
-    val keyName = displayName.replace(Regex("[ .]"), "_")
-      .lowercase()
+  private fun keyName(
+    displayName: String,
+    composePreviewSnapshotConfig: ComposePreviewSnapshotConfig? = null,
+  ): String {
+    val keyName = composePreviewSnapshotConfig?.let {
+      // If not default, we'll append the hashcode to the name to ensure uniqueness based on config
+      // Hashcode should preserve equality across field additions/removals in the future.
+      if (it.hashCode() != DEFAULT.hashCode()) {
+        "${displayName.normalize()}_${it.hashCode()}"
+      } else displayName.normalize()
+    } ?: displayName.normalize()
 
     if (keyName.length <= MAX_NAME_LENGTH) return keyName
-    // If the string is still too long, shorten to 32 characters
+    // If the string is still too long, shorten to max length
     return keyName.substring(0, MAX_NAME_LENGTH)
```

Comment:
Should we truncate the `displayName` portion of the filename first until the overall length is under our limit, otherwise we are cutting characters from our hash value which might cause issues with uniqueness.